### PR TITLE
Use the same flow architectures for lampe and sbi

### PR DESCRIPTION
**Pull request by francois-rozet**
This PR rewrites the flows of `npe_sbi` (using `nflows`) to match the ones of `npe_lampe`.

### CODE REVIEW

File: solvers/npe_sbi.py [hunk 1]
```diff
@@ -45,22 +46,46 @@ def set_objective(self, theta: Tensor, x: Tensor, prior: Distribution):
...
+
+            ts = []
+
+            for _ in range(self.transforms):
+                ts.extend([MAT(), transforms.ReversePermutation(features)])
+
+            transform = transforms.CompositeTransform(ts)
+            base = distributions.StandardNormal(shape=[features])
+
+            return flows.Flow(transform=transform, distribution=base)
```
Comment (JuliaLinhart) [inline solvers/npe_sbi.py:81]: Very nice for the autoregressive NSF! 
Did you check that all the parameters and the architecture align with the `lampe` flows? 
I think the default number of `hidden_features` is not the same for both.

File: solvers/npe_sbi.py [hunk 2]
```diff
@@ -45,22 +46,46 @@ def set_objective(self, theta: Tensor, x: Tensor, prior: Distribution):
...
 
         with dump():
             self.npe = npe.train(
-                validation_fraction=0.1,
+                validation_fraction=1 / len(self.theta),
                 max_num_epochs=n_iter + 1,
+                stop_after_epochs=n_iter + 1,
                 training_batch_size=128,
                 learning_rate=1e-3,
             )
```
Comment (JuliaLinhart) [inline solvers/npe_sbi.py:93]: Do I understand this right: the `stop_after_epochs` parameter makes sure we don't stop before the end of `n_iter + 1`, even if on the tiny "validation set" the loss will fluctuate a lot?

**Review by JuliaLinhart — COMMENTED** [review]
See comments in code.

File: solvers/npe_sbi.py (hunk 2 above)
Comment (francois-rozet) [reply solvers/npe_sbi.py:93]: Exactly, it's the same as ignoring the validation loss.

File: solvers/npe_sbi.py (hunk 1 above)
Comment (francois-rozet) [reply solvers/npe_sbi.py:81]: The default for lampe is 64 I think.

Comment (JuliaLinhart) [reply solvers/npe_sbi.py:81]: Yes ok, I just checked. I still think we should specify the `hidden_feature` explicitly, in case one wants to change that parameter.

Comment (francois-rozet) [reply solvers/npe_sbi.py:81]: I added `hidden_features=(64, 64) explicitly.`

**Comment by JuliaLinhart** [comment]
Ahh ok! @francois-rozet I though the size of the sequence had to match the `transforms` parameter. But instead the size of the sequence defines the number of layers right? which are called `blocks` in `sbi` .

**Comment by francois-rozet** [comment]
Yes, it's the number of hidden layers in the hyper network of EACH transformation.

**Comment by JuliaLinhart** [comment]
Perfect! We can merge this!

**Review by JuliaLinhart — APPROVED** [review]
